Improve handling of duplicate products

sssangha opened the pull request:
As illustrated in #233, when duplicate products are flagged (i.e. produced with same reference frame) in some instances the longer of the two may be discarded, which introduces a gap in the path. The proposed improvements compare the duplicate files and pass the longer of the two.

sssangha committed:
Specified download of compatible GDAL version

sssangha committed:
Kick out shorter of the duplicate scenes

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tools/ARIAtools/ARIAProduct.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6303352&bid=20498848)

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tools/ARIAtools/ARIAProduct.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6303353&bid=20498849)

sssangha commented:
@dbekaert codacy was vague in regards to how the code I introduced was complex, it just reporting some percentage without suggestions as we see for other cases regarding duplication and pythonic conventions.

Perhaps it would be appropriate to address general codacy issues in a separate PR and push this bug-fix through?

dbekaert commented:
The codacy issue is ok as it passed the PR standard for codacy. If its a severe issue it will report it.

To your question if we should in merge PR (in future) that fail Codacy and are bug fixes; would say no as the premise is for the PR to be up to coding standards we set before we merge it.

dbekaert commented on environment.yml:
```diff
@@ -10,7 +10,7 @@ channels:
   - python>=3.6
   - cartopy
-  - gdal>=3.0
+  - gdal=3.0.4
```
Se comment on other PR #234

sssangha committed:
Update environment with gdal range

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tools/ARIAtools/ARIAProduct.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6303352&bid=20498848)

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tools/ARIAtools/ARIAProduct.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6303353&bid=20498849)

sssangha commented:
> To your question if we should in merge PR (in future) that fail Codacy and are bug fixes; would say no as the premise is for the PR to be up to coding standards we set before we merge it.

I agree going forward, but in this case it isn't transparent how complexity was added. I was trying to say for minor cases where codacy doesn't fail, we could address in a separate PR. Code could be cleaned up as we did with RAiDER

sssangha committed:
Update environment attempt 2

With the previous update to query within a defined range of gdal versions, the program defaults to a newer version outside of the range.

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tools/ARIAtools/ARIAProduct.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6303353&bid=20498849)

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tools/ARIAtools/ARIAProduct.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6303352&bid=20498848)

dbekaert commented:
yes, agreed on the above; in case of sever codacy issues it would flag it. your PR still passed codacy, but had other issue on 
GDAL.

ehavazli reviewed (approved):
looks good to me
